Struct alignment code

mariacm12 opened the pull request:
## Description
Adding code for structure alignment following the sequence alignment pipeline. Process is described in HOWTO Notion page https://www.notion.so/asapdiscovery/HOWTO-Run-a-sequence-alignment-job-ae1bf71d092b44a9acfc3064362c3b99?pvs=4 

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [ ] Complete the sequence alignment pipeline

## Questions
- [ ] Question 1 ..

## Status
- [ ] Review needed


## Developers certificate of origin
- [ ] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

hmacdope commented on asapdiscovery-genetics/asapdiscovery/genetics/calculate_rmsd.py:
```diff
@@ -0,0 +1,151 @@
+    final_pdb: str,
+    target_chain="A",
+    ref_chain="A",
+):
```
If using types should add return value also
```suggestion
) -> Tuple[float, Path]:
```
You will need `from typing import Tuple` and `from pathlib import Path`

mariacm12 replied:
Thanks for the corrections! I think I will implement them locally and submit them as a separate commit. I made a few changes/additions to the files while running the job John requested and I want to avoid merge issues!

hmacdope replied:
Sounds good!

hmacdope commented on asapdiscovery-genetics/asapdiscovery/genetics/calculate_rmsd.py:
```diff
@@ -0,0 +1,151 @@
+    pdb_ref: str,
+    chain="A",
+    final_pdb="aligned_protein.pdb",
+):
```
Same here for return value type hints.

hmacdope commented on asapdiscovery-genetics/asapdiscovery/genetics/calculate_rmsd.py:
```diff
@@ -0,0 +1,151 @@
+            f"A folder with ColbFold results {results_dir} does not exist"
+        )
+
+    default_CF = "*_unrelaxed_rank_001_alphafold2_ptm_model_1_seed_*.pdb"
```
Make this a parameter that you can pass in with this default. Is there a way to make this simpler also? Or is this just what collabfold output looks like?

mariacm12 replied:
That's the default name format for ColabFold, at least for the run in the tutorial. But I can pass it as a parameter in the rare case the name changes.

hmacdope commented on asapdiscovery-genetics/asapdiscovery/genetics/calculate_rmsd.py:
```diff
@@ -0,0 +1,151 @@
+        return 0, ""
+    min_rmsd = np.argmin(rmsds)
+    min_rmsd_file = file_seed[min_rmsd]
+    print(f"Seed with less RMSD is {seeds[min_rmsd]} with RMSD {rmsds[min_rmsd]} A")
```
nitpick
```suggestion
    print(f"Seed with least RMSD is {seeds[min_rmsd]} with RMSD {rmsds[min_rmsd]} A")
```

hmacdope commented on asapdiscovery-genetics/asapdiscovery/genetics/calculate_rmsd.py:
```diff
@@ -0,0 +1,151 @@
+    return min_rmsd, str(final_pdb)
+
+
+def save_alignment_pymol(pdbs: list, labels: list, reference: str, session_save: str):
```
```suggestion
def save_alignment_pymol(pdbs: list, labels: list, reference: str, session_save: str) -> None:
```

hmacdope commented on asapdiscovery-genetics/asapdiscovery/genetics/scripts/structure_alignment.py:
```diff
@@ -0,0 +1,107 @@
+
+    aligned_pdbs = []
+    seq_labels = []
+    with open(seq_file) as f:
```
Should read a csv with something like `pandas.read_csv()` rather than by hand for more consistency and portability. This will handle everything for you. You can then just iterate over the resulting `DataFrame`

hmacdope reviewed (changes requested):
Amazing work @mariacm12! Just a few nitpicks.

mariacm12 commented:
@hmacdope 
I added some changes to the sequence alignment pipeline when addressing John's request on the SARS-CoV2 N protein (added them to this PR since it's still a draft). I added a couple of parameters to control the number of BLAST outputs and the option to select entries based on the host (I'll update the Notion page as well). There's a lot of input parameters now on the cli, so I don't know if it would sense to input these as a json or yaml file now. Let me know what you think!

hmacdope reviewed (approved):
All looks good to me @mariacm12! Merge when ready.

hmacdope commented:
Just need to fix the issues identified by `pre-commit` in failing check first.
